raven/processes/wps_generic_raster_subset.py

```diff
@@ -63,92 +49,85 @@ def __init__(self):
-                    nodata = raster_location['mini_raster_nodata']
-
-                    aff = Affine(*grid_properties)
-
-                    LOGGER.info('Writing raster data to {}'.format(raster_subset))
-
-                    masked_array = np.ma.masked_values(raster, nodata)
-                    if masked_array.mask.all():
-                        msg = 'Subset {} is empty, continuing...'.format(i)
-                        LOGGER.warning(msg)
-
-                    normal_array = np.asarray(masked_array, dtype=data_type)
-
-                    # Write to GeoTIFF
-                    with rio.open(raster_subset, 'w', driver='GTiff', count=1, compress=raster_compression,
-                                  height=raster.shape[0], width=raster.shape[1], dtype=data_type, transform=aff,
-                                  crs=vec_crs or ras_crs, nodata=nodata) as f:
-                        f.write(normal_array, 1)
-
-                except Exception as e:
-                    msg = 'Failed to write raster outputs: {}'.format(e)
-                    LOGGER.error(msg)
-                    raise Exception(msg)
-
-            # `shutil.make_archive` could potentially cause problems with multi-thread? Worth investigating later.
-            out_fn = os.path.join(self.workdir, self.identifier)
-            shutil.make_archive(base_name=out_fn, format='zip', root_dir=out_dir, logger=LOGGER)
-
-            response.outputs['raster'].file = '{}.zip'.format(out_fn)
+                vector_file,
+                raster_file,
+                band=band,
+                all_touched=touches,
+                raster_out=True,
+            )
+
+            raster_files = zonalstats_raster_file(
+                stats,
+                working_dir=self.workdir,
+                data_type=data_type,
+                crs=vec_crs or ras_crs,
+            )
+
+            if len(raster_files) > 1:
+                ml = MetaLink4(
+                    "test-ml-1",
+                    "MetaLink with links to raster files.",
+                    workdir=self.workdir,
+                )
+                for i, file in enumerate(raster_files):
+                    # Create a MetaFile instance, which instantiates a ComplexOutput object.
+                    mf = MetaFile(
+                        file.name, description="Raster file", fmt=FORMATS.GEOTIFF
+                    )
+                    mf.file = (
+                        file.as_posix()
+                    )  # or mf.file = <path to file> or mf.url = <url>
+                    ml.append(mf)
+
+                response.outputs["raster"].data = ml.xml
+            else:
+                response.outputs["raster"].file = raster_files[0]
 
         except Exception as e:
-            msg = 'Failed to perform raster subset using {} and {}: {}'.format(shape_url, raster_url, e)
+            msg = f"Failed to perform raster subset using {shape_url}{f' and {raster_url} ' if not None else ''}: {e}"
```

Comment:
An f-string inside an f-string is hard to read I find.. and in this case I think it should be:
```python
f"Failed to perform raster subset using {shape_url}{f' and {raster_url} ' if raster_url else ''}: {e}"
```